core/vm/evm.go

```diff
@@ -463,13 +463,30 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 
 	ret, err := run(evm, contract, nil, false)
 
-	// check whether the max code size has been exceeded
-	maxCodeSizeExceeded := evm.chainRules.IsEIP158 && len(ret) > params.MaxCodeSize
+	// Check whether the max code size has been exceeded, assign err if the case.
+	if err == nil && evm.chainRules.IsEIP158 && len(ret) > params.MaxCodeSize {
+		err = ErrMaxCodeSizeExceeded
+	}
+
+	if err == nil {
+		isEIP3541 := func() bool {
+			for _, eip := range evm.vmConfig.ExtraEips {
+				if eip == 3541 {
+					return true
+				}
+			}
+			return false
+		}()
```

Comment:
I'd just do `evm.chainRules.IsEip3541` -- and later on change it to `IsLondon` or whatever

Reply:
I only refactored this. We are happy that we can create tests for `Berlin+3541` for this.

Reply:
I only refactored this. We are happy that we can create tests for `Berlin+3541` for this.

Reply:
Yeah, that's fine, but in the long run we should have a proper config and tie it to a forkname instead. That ^ is a bit of a hack